effects/kill_effect.py

```diff
@@ -0,0 +1,42 @@
+from game import Game
+from effects.effect import Effect
+from citizens.citizen import Citizen
+from citizens.player import Player
+import user_interaction
+import utils
+
+
+class KillEffect(Effect):
+    def __init__(self, name: str, game: Game, creator: Citizen):
+        Effect.__init__(self, name, game, creator)
+
+    def activate(self, target_number):
+        target_number = user_interaction.read_index('config.type_target')
+        while(not self.validate(target_number)):
+            user_interaction.show_active_instant("config.type_error")
+            target_number = user_interaction.read_index("config.type_target")
+
+        self.target = self.game.citizens[target_number - 1]
+
+    def resolve(self):
+
+        if isinstance(self.target, Player):
+            self.target.hp -= 1
+            user_interaction.save_passive('config.lose_hp')
+            user_interaction.save_global('config.failed_kill')
+
+        else:
+            self.target.hp -= 1
+            user_interaction.save_global('config.citizen_die')
+            if self.target.citizen_card is not None:
+                self.creator.stolen_cards.append(self.target.citizen_card)
+                self.target.citizen_card = None
+                user_interaction.save_active('config.card_stolen')
+            else:
+                user_interaction.save_active('config.no_card')
+
+    def on_clear(self):
+        pass
+
+    def validate(self, target_number):
```

Comment:
Please, use type hints for method arguments and method return values

def validate(self, target_number: int) -> None:

Rewrite other methods in this style.

Reply:
Done